src/components/CustomDrawer.tsx

```diff
@@ -0,0 +1,152 @@
+/* eslint-disable react-native-a11y/has-valid-accessibility-ignores-invert-colors */
+import React from 'react';
+import { View, Text, TouchableOpacity, Image, ScrollView } from 'react-native';
+import { LinearGradient } from 'expo-linear-gradient';
+import { useNavigation } from '@react-navigation/native';
+import { useMyEditableWorkouts } from '../hooks/useMyEditableWorkouts';
+import { useAuth } from '../contexts/AuthProvider';
+import { MaterialCommunityIcons } from '@expo/vector-icons';
+import DrawerWorkoutBlock from './blocks/DrawerWorkoutBlocks';
+import { Ionicons } from '@expo/vector-icons';
+import { getUserProfile } from '../utils/db/profiles';
+import { ProfileType } from '../types';
+
+const CustomDrawer = () => {
+  const navigation = useNavigation();
+  const { session } = useAuth();
+  const { editableWorkouts, fetch } = useMyEditableWorkouts(session);
+  const [userData, setUserData] = React.useState<ProfileType>({});
+
+  //loads user profile data
+  React.useEffect(() => {
+    const fetchProfile = async () => {
+      const data = await getUserProfile(session);
+      setUserData(data);
+    };
+
+    fetchProfile().catch(console.error);
+  }, []);
+
+  /**
+   * on refresh, fetches profile data
+   * + editable workouts and rerenders
+   */
+  const refresh = () => {
+    if (fetch != undefined) {
+      fetch();
+    }
+    getUserProfile(session)
+      .then((data) => {
+        setUserData(data);
+      })
+      .catch(console.error);
+  };
+
+  return (
+    <View className="h-full w-full">
+      {/**Banner + Profile Block */}
+      <LinearGradient className="mb-3 h-32" colors={['#888787', '#9A9A9A', '#FFFFFF']}>
```

Comment:
I think the linear gradient obfuscates the UI rather than enhances it. It could be nice to incorporate this elsewhere but I don't think it looks the greatest with this profile header